### PR TITLE
bpo-47061: document module deprecations due to PEP 594

### PR DESCRIPTION
**Pull request by brettcannon**
Also removed asynchat, asyncore, and smtpd from their respective toctree entries so they are only  in the superceded subtree.

https://bugs.python.org/issue47061

Automerge-Triggered-By: GH:brettcannon

### CODE REVIEW

**Review by tiran — APPROVED** [review]

File: Doc/library/chunk.rst [hunk 1]
```diff
@@ -16,6 +17,9 @@
    single: Real Media File Format
    single: RMFF
 
+.. deprecated:: 3.11
+   The :mod:`chunk` module is deprecated (see :pep`594` for details).
```
Comment (tiran) [inline Doc/library/chunk.rst]: ``
```suggestion
   The :mod:`chunk` module is deprecated (see :pep:`594` for details).
```

File: Doc/library/audioop.rst [hunk 2]
```diff
@@ -3,6 +3,10 @@
 
 .. module:: audioop
    :synopsis: Manipulate raw audio data.
+   :deprecated:
+
+.. deprecated:: 3.11
+   The :mod:`audioop` is deprecated (see :pep:`594` for details).
```
Comment (ngnpope) [inline Doc/library/audioop.rst]:
```suggestion
   The :mod:`audioop` module is deprecated (see :pep:`594` for details).
```

File: Misc/NEWS.d/next/Library/2022-03-18-13-30-40.bpo-47061.etLHK5.rst [hunk 3]
```diff
@@ -0,0 +1,7 @@
+Deprecate the various modules listed by :pep:`594`:
+
+- aifc - asynchat: https://bugs.python.org/issue47022 - asyncore:
+https://bugs.python.org/issue47022 - audioop - cgi - cgitb - chunk - crypt -
+imghdr - msilib - nntplib - nis - ossaudiodev - pipes - smtpd:
+https://bugs.python.org/issue47022 - sndhdr - spwd - sunau - telnetlib - uu
+- xdrlib
```
Comment (hugovk) [inline Misc/NEWS.d/next/Library/2022-03-18-13-30-40.bpo-47061.etLHK5.rst]: Docs build is failing:

```
Warning, treated as error:
../build/NEWS:59:Bullet list ends without a blank line; unexpected unindent.
make: *** [Makefile:53: build] Error 2
##[error]Bash exited with code '2'.
```

Removing this initial hyphen is one fix:

```suggestion
aifc - asynchat: https://bugs.python.org/issue47022 - asyncore:
https://bugs.python.org/issue47022 - audioop - cgi - cgitb - chunk - crypt -
imghdr - msilib - nntplib - nis - ossaudiodev - pipes - smtpd:
https://bugs.python.org/issue47022 - sndhdr - spwd - sunau - telnetlib - uu
- xdrlib
```

Will produce a NEWS file containing:

```
- bpo-47061: Deprecate the various modules listed by :pep:`594`:

  aifc - asynchat: https://bugs.python.org/issue47022 - asyncore:
  https://bugs.python.org/issue47022 - audioop - cgi - cgitb - chunk - crypt
  - imghdr - msilib - nntplib - nis - ossaudiodev - pipes - smtpd:
  https://bugs.python.org/issue47022 - sndhdr - spwd - sunau - telnetlib -
  uu - xdrlib
```


(Reminder about https://github.com/python/cpython/pull/30419 to upload a generated NEWS file as part of the build.)

**Comment by hugovk** [comment]
Should this also have an entry in "What's New In Python 3.11"?

**Comment by brettcannon** [comment]
> Should this also have an entry in "What's New In Python 3.11"?

I purposefully left one out as I will updating it as I go along and deprecate the modules which I will be doing in separate PRs.

**Comment by brettcannon** [comment]
Thanks everyone for the reviews!

**Comment by hugovk** [comment]
And thanks for the PR!

Is a manual 3.9 backport needed?

**Comment by brettcannon** [comment]
@hugovk yep, a manual backport for 3.9 is needed. I was planning to tackle it probably this Friday, but if you want to take it on I will not stop you. 😉

**Comment by hugovk** [comment]
There we go, and the `cherry_picker 9ac2de922a0f783bd43b8e026e4fb70fd1888572 3.9` command applied cleanly first time :)

**Pull request merged**